apps/api/v2/src/modules/slots/slots-2024-09-04/services/slots-input.service.ts

```diff
@@ -0,0 +1,68 @@
+import { EventTypesRepository_2024_06_14 } from "@/ee/event-types/event-types_2024_06_14/event-types.repository";
+import { UsersRepository } from "@/modules/users/users.repository";
+import { Injectable, NotFoundException } from "@nestjs/common";
+import { DateTime } from "luxon";
+
+import { dynamicEvent } from "@calcom/platform-libraries";
+import { GetSlotsInput_2024_09_04 } from "@calcom/platform-types";
+
+@Injectable()
+export class SlotsInputService_2024_09_04 {
+  constructor(
+    private readonly eventTypeRepository: EventTypesRepository_2024_06_14,
+    private readonly usersRepository: UsersRepository
+  ) {}
+
+  async transformGetSlotsQuery(query: GetSlotsInput_2024_09_04) {
+    const eventType = await this.getEventType(query);
+    if (!eventType) {
+      throw new NotFoundException(`Event Type not found`);
+    }
+    const isTeamEvent = !!eventType?.teamId;
+
+    const startTime = query.start;
+    const endTime = this.adjustEndTime(query.end);
+    const duration = query.duration;
+    const eventTypeId = eventType.id;
+    const eventTypeSlug = eventType.slug;
+    const usernameList = "usernames" in query ? query.usernames : [];
+    const timeZone = query.timeZone;
+    const orgSlug = "organizationSlug" in query ? query.organizationSlug : null;
+
+    return {
+      isTeamEvent,
+      startTime,
+      endTime,
+      duration,
+      eventTypeId,
+      eventTypeSlug,
+      usernameList,
+      timeZone,
+      orgSlug,
+    };
+  }
+
+  private async getEventType(input: GetSlotsInput_2024_09_04) {
+    if ("eventTypeId" in input) {
+      return this.eventTypeRepository.getEventTypeById(input.eventTypeId);
+    }
+    if ("eventTypeSlug" in input) {
+      const user = await this.usersRepository.findByUsername(input.username);
+      if (!user) {
+        throw new NotFoundException(`User with username ${input.username} not found`);
+      }
+      return this.eventTypeRepository.getUserEventTypeBySlug(user.id, input.eventTypeSlug);
```

Comment:
Adressed [concern](https://github.com/calcom/cal.com/pull/16489#discussion_r1775017431) of Syed in original PR.